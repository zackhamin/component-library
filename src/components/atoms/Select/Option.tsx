interface OptionProps {
	children: React.ReactNode;
	value: string | number;
	isDisabled?: boolean;
}

const Option = ({ children, value, isDisabled }: OptionProps) => {
	return (
		<option value={value} disabled={isDisabled}>
			{children}
		</option>
	);
};

export { Option };
