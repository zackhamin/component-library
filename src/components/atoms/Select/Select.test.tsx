import { Option } from './Option';
import { Select } from '.';
import { render, screen } from '@testing-library/react';

const defaultProps = {
	testId: 'selectTestId',
	id: 'select',
	class: 'class-prop',
	name: 'test-name',
	children: (
		<>
			<Option value="select-option">Select Option</Option>
			<Option value="wrong-item">Wrong Item in Parcel</Option>
		</>
	),
};

describe('Select', () => {
	it('renders correctly', () => {
		const { container } = render(<Select {...defaultProps} />);
		expect(container).toMatchSnapshot();
	});

	it('should use options prop to populate dropdown', () => {
		render(<Select {...defaultProps} />);
		expect(screen.getByText('Select Option')).toBeInTheDocument();
		expect(screen.getByDisplayValue(/select option/i));
		expect(screen.getByText('Wrong Item in Parcel')).toBeInTheDocument();
	});

	it('adds ID, class and name to element', () => {
		const changeHandler = jest.fn();
		render(<Select {...defaultProps} onChange={changeHandler} />);
		expect(screen.getByRole('combobox')).toHaveAttribute('id', 'select');
		expect(screen.getByTestId('selectTestId')).toBeInTheDocument();
		expect(screen.getByText('Wrong Item in Parcel')).toBeInTheDocument();
		expect(screen.getByRole('combobox')).toHaveAttribute('name', 'test-name');
	});

	it('should display an SVG chevron', () => {
		render(<Select {...defaultProps} />);
		expect(screen.getAllByTestId('CHEVRON')).toHaveLength(1);
	});

	it('should have disabled attribute when prop passed', () => {
		render(<Select {...defaultProps} isDisabled={true} />);
		expect(screen.getByRole('combobox')).toHaveAttribute('disabled');
	});

	it('renders with placeholder option', () => {
		render(<Select {...defaultProps} placeholder="Please select an option" />);

		const placeholderOption = screen.getAllByRole('option')[0];

		expect(placeholderOption).toHaveTextContent('Please select an option');
		expect(placeholderOption).toHaveAttribute('disabled');
	});
});
