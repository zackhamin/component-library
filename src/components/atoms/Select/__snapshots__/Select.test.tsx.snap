// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select renders correctly 1`] = `
.c0 {
  background: #ffffff;
  box-sizing: border-box;
  font-size: 1rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  overflow: hidden;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  position: relative;
}

.c0.select-large {
  width: 100%;
}

.c0.select-small {
  width: 70px;
}

.c1 {
  position: absolute;
  right: 15px;
  top: 50%;
  z-index: 3;
  -webkit-transform: translateY(-50%);
  -ms-transform: translateY(-50%);
  transform: translateY(-50%);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

<div>
  <div
    class="c0 select-large"
  >
    <div
      class="c1"
    >
      <svg
        aria-hidden="true"
        color="#00594C"
        data-testid="CHEVRON"
        focusable="false"
        font-size="1.4rem"
        height="1em"
        viewBox="0 0 14 8"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M.293.293a1 1 0 0 1 1.414 0L7 5.586 12.293.293a1 1 0 1 1 1.414 1.414l-6 6a1 1 0 0 1-1.414 0l-6-6a1 1 0 0 1 0-1.414Z"
          fill="currentColor"
        />
      </svg>
    </div>
    <select
      class="class-prop"
      data-testid="selectTestId"
      id="select"
      name="test-name"
    >
      <option
        value="select-option"
      >
        Select Option
      </option>
      <option
        value="wrong-item"
      >
        Wrong Item in Parcel
      </option>
    </select>
  </div>
</div>
`;
